fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/AbstractImportHandler.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.bulkimport.importhandler;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.poi.ss.usermodel.*;
+import org.joda.time.LocalDate;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Iterator;
+
+public abstract class AbstractImportHandler implements ImportHandler {
+
+    protected Integer getNumberOfRows(Sheet sheet, int primaryColumn) {
+        Integer noOfEntries = 1;
+        // getLastRowNum and getPhysicalNumberOfRows showing false values
+        // sometimes
+        while (sheet.getRow(noOfEntries) !=null && sheet.getRow(noOfEntries).getCell(primaryColumn) != null) {
+            noOfEntries++;
+        }
+
+        return noOfEntries;
+    }
+
+    protected boolean isNotImported(Row row, int statusColumn) {
+        return !readAsString(statusColumn, row).equals("Imported");
+    }
+
+    protected Long readAsLong(int colIndex, Row row) {
+        try {
+            Cell c = row.getCell(colIndex);
+            if (c == null || c.getCellType() == Cell.CELL_TYPE_BLANK)
+                return null;
+            FormulaEvaluator eval = row.getSheet().getWorkbook().getCreationHelper().createFormulaEvaluator();
+            if(c.getCellType() == Cell.CELL_TYPE_FORMULA) {
+                CellValue val = null;
+                try {
+                    val = eval.evaluate(c);
+                } catch (NullPointerException npe) {
+                    return null;
+                }
+                return ((Double) val.getNumberValue()).longValue();
+            }
+            return ((Double) c.getNumericCellValue()).longValue();
+        } catch (RuntimeException re) {
+            return Long.parseLong(row.getCell(colIndex).getStringCellValue());
```

Comment:
if cell value is not numeric then in return it will throw exception. Handle the same scenario in other places also.